### PR TITLE
BUG: accept zeros on numpy.random dirichlet function

### PR DESCRIPTION
**Pull request by pcralmeida**
Changed `alpha` value error to pass a null value. This way, dirichlet function wont raise a value exception at 0. Also added new `test_dirichlet` function on `numpy.tests.test_matlib.py`.

### CODE REVIEW

**Comment by MatteoRaso** [comment]
Thanks for the PR. Just to let everybody know, this is in reference to #22547.

File: numpy/tests/test_matlib.py [hunk 1]
```diff
@@ -56,3 +57,6 @@ def test_repmat():
     y = np.array([[0, 1, 2, 3, 0, 1, 2, 3],
                   [0, 1, 2, 3, 0, 1, 2, 3]])
     assert_array_equal(x, y)
+
+def test_dirichlet():
+    dirichlet([5, 9, 0, 8])
```
Comment (MatteoRaso) [inline numpy/tests/test_matlib.py:62]:
```suggestion
def test_dirichlet():
    y = dirichlet([5, 9, 0, 8])
    assert y[2] == 0
```
We don't just want to that the function doesn't throw an error, we also want to show that the function returns the right result when 0 is given as input.

**Review by seberg — CHANGES_REQUESTED** [review]
Thanks, but all changes here need to be moved around, they should be OK changes, but they are at the wrong spot.

File: numpy/random/mtrand.pyx [hunk 2]
```diff
@@ -4506,8 +4506,8 @@ cdef class RandomState:
         alpha_arr = <np.ndarray>np.PyArray_FROMANY(
             alpha, np.NPY_DOUBLE, 1, 1,
             np.NPY_ARRAY_ALIGNED | np.NPY_ARRAY_C_CONTIGUOUS)
-        if np.any(np.less_equal(alpha_arr, 0)):
-            raise ValueError('alpha <= 0')
+        if np.any(np.less(alpha_arr, 0)):
+            raise ValueError('alpha < 0')
```
[the remark this answers is not in the document]
Comment (pcralmeida) [reply numpy/random/mtrand.pyx:4510]: I've added the changes to `generator.pyx` as intended, and also passed the test to the `numpy/random` submodule. However, the test would raise the `alpha = 0` value error from `mtrand.pyx`, so I maintained the changes that were made. The changes now seem OK, but I will wait for your feedback.

[the remark this answers is not in the document]
Comment (seberg) [reply numpy/random/mtrand.pyx:4510]: The value-error is desired in `mtrand` (strict compatibility).  The tests just need to be different.  We should make sure that generator is tested, not `mtrand` (i.e. not `np.random.diriclet` but `np.random.default_rng().dirichlet`.  Those tests live in a different file.

In a sense, we want to nudge users towards preferring `np.random.default_rng()` anyway, so there is probably not much reason to improve mtrand.

[the remark this answers is not in the document]
Comment (pcralmeida) [reply numpy/random/mtrand.pyx]: Hey, sorry for the long wait. I've read your comment and I've reverted the changes in `mtrand`. Also fixed the test having `np.random.default_rng()` in account. Hope these changes are good now.

**Review by MatteoRaso — APPROVED** [review]
LGTM.

**Review by seberg — APPROVED** [review]

**Comment by seberg** [comment]
Thanks for the followup, looks good now.

File: numpy/random/tests/test_randomstate.py [hunk 3]
```diff
@@ -812,6 +812,10 @@ def test_dirichlet_bad_alpha(self):
         alpha = np.array([5.4e-01, -1.0e-16])
         assert_raises(ValueError, random.dirichlet, alpha)
 
+    def test_dirichlet_zero_alpha(self):
+        y = random.default_rng().dirichlet([5, 9, 0, 8])
```
Comment (WarrenWeckesser) [inline numpy/random/tests/test_randomstate.py:816]: This is using the result of `default_rng()`, which is a `Generator` instance (not `RandomState`), so this test should not be in this file.  I'll fix this in the pull request I am working on for https://github.com/numpy/numpy/issues/24210.

**Pull request merged**